src/Time/Formatting.hs

```diff
@@ -52,7 +81,11 @@ instance Series ('[] :: [Rat]) where
     seriesF :: Time someUnit -> String
     seriesF _ = ""
 
-instance (KnownRatName unit, Series units)
+instance ( KnownRatName unit, Series units
+#if ( __GLASGOW_HASKELL__ >= 804 )
+         , IsDescending (unit : units) ~ 'True
```

Comment:
Okay, here comes the most interesting part. As you already described in documentation, the type error user will see contains message like this:

```haskell
Couldn't match type ‘'False’ with ‘'True’
```

This error message is not actually very useful. I think it would be better to return some custom `ErrorMessage` if `IsDescending ~ 'False` like you did in `universum`:

* https://github.com/serokell/universum/blob/2c524b5ddefad5b21232aee59fcb6caf8ff1c9f6/src/Universum/Container/Class.hs#L697

Reply:
I have no idea how is that working, so I won't be able to implement that, sorry, it's too magical to me to understand.

Reply:
@vrom911 Don't worry about that :) I think only 1% of Haskellers (or even less) is able to implement that. And I wasn't able as well until some recent times. I agree, this is indeed to magical... I will do that after rest of your changes.